models/users.js

```diff
@@ -145,9 +145,15 @@ const fetchPaginatedUsers = async (query) => {
     // INFO: https://github.com/Real-Dev-Squad/website-backend/pull/873#discussion_r1064229932
     const size = parseInt(query.size) || 100;
     const doc = (query.next || query.prev) && (await userModel.doc(query.next || query.prev).get());
-
-    let dbQuery = userModel.where("roles.archived", "==", false).orderBy("username");
-
+    let dbQuery;
+    if (query.q) {
+      const [queryName, queryValue] = query.q.split(":");
+      if (queryName === "includes" && queryValue === "archived") {
+        dbQuery = userModel;
+      }
+    } else {
```

Comment:
Is this a specific use case we are solving here, could we consider generalizing it.
We would receive different queries right in the future, could we think of a better way here?
I am also thinking about it, will update once I find relevant document

Reply:
Have you looked in the repo, for the cases similar to this?

Reply:
So Ankush suggested this API structure go ahead with
![image](https://github.com/Real-Dev-Squad/website-backend/assets/75560307/7155a190-5f8d-4c06-b102-69ce1e1781d4)

Regarding the generalizing will make the changes

Reply:
Done added the changes

Reply:
please add test cases to validate with this params, when this params are absent as well, rest looks good

Reply:
have added the test case for this new scenario